spec/unit/events_collection/lockfile_spec.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../../../lib/events_collection/lockfile'
+
+describe 'Lockfile' do
+  let(:basepath) { '/tmp' }
+  let(:path) { File.join('/tmp', 'events_collection_run.lock') }
+  let(:new_lockfile) { CommonEvents::Lockfile.new(basepath) }
+
+  before(:each) do
+    File.delete(path) if File.exist?(path)
+  end
+
+  it 'sets the correct lockfile path' do
+    filepaths = ['/path/path', '/path/', '/path', 'path', '']
+    filepaths.each do |filepath|
+      expect(CommonEvents::Lockfile.new(filepath).filepath).to eq(File.join(filepath, '/events_collection_run.lock'))
+      File.delete(filepath) if File.exist?(filepath)
+    end
+  end
+
+  it 'requires a lockfile path' do
+    expect { CommonEvents::Lockfile.new }.to raise_error(ArgumentError)
+  end
+
+  it 'creates the lockfile with the correct format' do
+    new_lockfile.write_lockfile
+    expect(File.exist?(path)).to be true
+    lockfile_contents = JSON.parse(File.read(path))
+    expect(lockfile_contents['pid']).to be_an(Integer)
+    expect(lockfile_contents['program_name']).to be_a(String)
+  end
+
+  it 'checks if the lockfile exists' do
+    lockfile = new_lockfile
+    expect(lockfile.lockfile_exists?).to be false
+    lockfile.write_lockfile
+    expect(lockfile.lockfile_exists?).to be true
+  end
+
+  it 'can delete the lockfile' do
+    lockfile = new_lockfile
+    lockfile.write_lockfile
+    expect(File.exist?(path)).to be true
+    lockfile.remove_lockfile
+    expect(File.exist?(path)).to be false
+  end
+
+  it 'returns lockfile info correctly' do
+    lockfile = new_lockfile
+    expect(lockfile.info).to eql({ pid: nil, program_name: '' })
+    lockfile.write_lockfile
+    expect(lockfile.info['pid']).to be_an(Integer)
+    expect(lockfile.info['program_name']).to be_a(String)
+    File.delete(path)
+    expect(lockfile.info).to eql({ pid: nil, program_name: '' })
+  end
```

Comment:
Can we add a test for what happens if validate command finds that the program name doesn't match? To ensure that the rescue block in validate_command fires, removes the lock file and allows the new one to be created?

Reply:
Same with this comment. I have no idea why this is attached to this line! 🤣